xchange-coinmate/src/main/java/org/knowm/xchange/coinmate/CoinmateAdapters.java

```diff
@@ -338,4 +336,50 @@ public static String adaptSortOrder(TradeHistoryParamsSorted.Order order) {
         throw new IllegalArgumentException();
     }
   }
+
+  public static List<Order> adaptOrders(CoinmateOrders coinmateOrders) {
+    List<Order> ordersList = new ArrayList<>(1);
+
+    CoinmateOrder entry = coinmateOrders.getData();
+
+    Order.OrderType orderType;
+
+    if ("BUY".equals(entry.getType())) {
+      orderType = Order.OrderType.BID;
+    } else if ("SELL".equals(entry.getType())) {
+      orderType = Order.OrderType.ASK;
+    } else {
+      throw new CoinmateException("Unknown order type");
+    }
+    Order.OrderStatus orderStatus;
+    if ("CANCELLED".equals(entry.getStatus())) {
+      orderStatus = Order.OrderStatus.CANCELED;
+    } else if ("FILLED".equals(entry.getStatus())) {
+      orderStatus = Order.OrderStatus.FILLED;
+    } else if ("PARTIALLY_FILLED".equals(entry.getStatus())) {
+      orderStatus = Order.OrderStatus.PARTIALLY_FILLED;
+    } else if ("OPEN".equals(entry.getStatus())) {
+      orderStatus = Order.OrderStatus.NEW;
+    } else {
+      orderStatus = Order.OrderStatus.UNKNOWN;
+    }
+
+    // TODO: we can probably use `orderTradeType` to distinguish between Market and Limit order
+    Order order =
+        new MarketOrder(orderType,
+            entry.getOriginalAmount(),
+            null,
+            Long.toString(entry.getId()),
+            new Date(entry.getTimestamp()),
+            entry.getAvgPrice(),
+            entry.getOriginalAmount().subtract(entry.getRemainingAmount()),
+            null,
+            orderStatus,
+            null);
+
+
+    ordersList.add(order);
+
+    return ordersList;
```

Comment:
Not necessary but a bit more elegant you can do something like 

`return Collections.singletonList(new MarketOrder(orderType,
            entry.getOriginalAmount(),
            null,
            Long.toString(entry.getId()),
            new Date(entry.getTimestamp()),
            entry.getAvgPrice(),
            entry.getOriginalAmount().subtract(entry.getRemainingAmount()),
            null,
            orderStatus,
            null));`

Reply:
Thanks